### PR TITLE
Set default value for host and share

### LINKED ISSUE #40170: SambaHook unable to join path when Share attribute is None
### Apache Airflow version

2.9.2

### If "Other Airflow 2 version" selected, which one?

_No response_

### What happened?

SambaHook encounters an error when Share attribute is intended to be a blank value.

This is the SambaHook as an environment variable without a schema. The schema is relabeled as share.

```
AIRFLOW_CONN_SAMBA='samba://username:password@hostname'
```

Because the schema is not set, the function _join_path encounters an error.

```
>>> from airflow.models import Connection
>>> from airflow.providers.samba.hooks.samba import SambaHook
>>> hook = SambaHook(samba_conn_id="samba")
>>> hook._join_path("test")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.11/site-packages/airflow/providers/samba/hooks/samba.py", line 87, in _join_path
    return f"//{posixpath.join(self._host, self._share, path.lstrip('/'))}"
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen posixpath>", line 90, in join
  File "<frozen genericpath>", line 152, in _check_arg_types
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'NoneType'
```

A blank share attribute will not work either since the value falls back to the connection's schema.

```
>>> hook = SambaHook(samba_conn_id="samba", share="")
[2024-06-11T07:16:25.988+0000] {base.py:83} INFO - Using connection ID 'samba' for task execution.
>>> hook._join_path("test")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.11/site-packages/airflow/providers/samba/hooks/samba.py", line 87, in _join_path
    return f"//{posixpath.join(self._host, self._share, path.lstrip('/'))}"
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen posixpath>", line 90, in join
  File "<frozen genericpath>", line 152, in _check_arg_types
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'NoneType'
```

Only after setting `_share` as a blank does the function run as expected.

```
>>> print(hook._share)
None
>>> hook._share=""
>>> hook._join_path("test")
'//hostname/test'
```

### What you think should happen instead?

The `_join_path` function should check whether `_share` is `None` before calling `posixpath.join`.

### How to reproduce

This is slightly difficult to recreate in a fresh Airflow instance since a hook cannot be initialized with an existing connection object. The issue is when the connection object with `None` values  is retrieved from the hook's `__init__` function which is used to set the hook's attributes

```
from airflow.models import Connection
from airflow.providers.samba.hooks.samba import SambaHook
conn = Connection(conn_id="test", conn_type="samba", host="hostname")
```

The `_host` and `_schema` values can be override to simulate the behaviour.

```
from airflow.models import Connection
from airflow.providers.samba.hooks.samba import SambaHook
hook = SambaHook(samba_conn_id="samba")
hook._scheme = None

hook._join_path("test")
```

### Operating System

n/a

### Versions of Apache Airflow Providers

`apache-airflow-providers-samba==4.7.0`

### Deployment

Astronomer

### Deployment details

_No response_

### Anything else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by wolfier**
Set host and share attribute values to a blank string as other class functions expect those attributes to have values other than None.

closes: #40170

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by hussein-awala — COMMENTED** [review]
Could you add a test to validate your bug fix?

File: airflow/providers/samba/hooks/samba.py [hunk 1]
```diff
@@ -59,8 +59,8 @@ def __init__(self, samba_conn_id: str = default_conn_name, share: str | None = N
 
         connection_cache: dict[str, smbprotocol.connection.Connection] = {}
 
-        self._host = conn.host
-        self._share = share or conn.schema
+        self._host = conn.host or ""
+        self._share = share or conn.schema or ""
```
Comment (amoghrajesh) [inline airflow/providers/samba/hooks/samba.py:64]: Can you please add a unit test for this?

Comment (wolfier) [reply airflow/providers/samba/hooks/samba.py:64]: Done!

File: tests/providers/samba/hooks/test_samba.py [hunk 2]
```diff
@@ -35,6 +35,12 @@
     password="password",
 )
 
+NULL_HOST_SCHEMA_CONNECTION = Connection(
+    host=None,
+    schema=None,
+    login="username",
+    password="password",
+)
```
Comment (eladkal) [inline tests/providers/samba/hooks/test_samba.py:43]:
```suggestion
)

```

To fix static checks

**Pull request closed without merging**